assume/strategies/learning_strategies.py

```diff
@@ -60,6 +61,10 @@ def __init__(self, *args, **kwargs):
         # sets the device of the actor network
         device = kwargs.get("device", "cpu")
         self.device = th.device(device if th.cuda.is_available() else "cpu")
+        if self.learning_mode and not self.learning_role:
+            raise AssumeException("Learning Role must be set in LearningMode")
```

Comment:
The check to fail early here was added by myself and is required, as the register method is called afterwards anyways on the self.learning_role.
So we need to adjust the tests here to add a mock learning_role when creating learningStrategies..?

This reduces testability slightly, as we have the dependency on the learning role - but I think it is better to fail early.
Alternative would be to disable registration if learning_role is not set, which could be silent unwanted behavior as well.